x/slashing/keeper/signing_info.go

```diff
@@ -158,3 +159,33 @@ func (k Keeper) clearValidatorMissedBlockBitArray(ctx sdk.Context, address sdk.C
 		store.Delete(iter.Key())
 	}
 }
+
+// RemoveValidatorSigningInfo removes the validator signing info from a consensus address key
+func (k Keeper) RemoveValidatorSigningInfo(ctx sdk.Context, address sdk.ConsAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.ValidatorSigningInfoKey(address))
+}
+
+func (k Keeper) PerformConsensusPubKeyUpdate(ctx sdk.Context, oldPubKey cryptotypes.PubKey, newPubKey cryptotypes.PubKey) error {
+	// Connect new consensus address with PubKey
+	if err := k.AddPubkey(ctx, newPubKey); err != nil {
+		return err
+	}
+
+	// Migrate ValidatorSigningInfo from oldPubKey to newPubKey
+	signingInfo, found := k.GetValidatorSigningInfo(ctx, sdk.ConsAddress(oldPubKey.Address()))
+	if !found {
+		return nil
```

Comment:
Shouldn't we error if the old key isn't found? This is the behavior I'd expect

Reply:
Oops my bad, thanks for pointing.